keras_spiking/layers.py

```diff
@@ -535,6 +535,11 @@ class LowpassCell(KerasSpikingCell):
         then we often don't need to filter them either).
     level_initializer : str or ``tf.keras.initializers.Initializer``
         Initializer for filter state.
+    initial_level_constraint : str or ``tf.keras.constraints.Constraint``
+        Constraint for ``initial_level``.
+    tau_var_constraint : str or ``tf.keras.constraints.Constraint``
```

Comment:
I'd just call this `tau_constraint` (to align with the other parameter in init here, which is just `tau`).

Reply:
I was following this pattern: https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/python/keras/layers/core.py#L1191-L1198 where it seems to be common for it to be named after the weights attribute that it belongs to (which is `tau_var` and not `tau`).

Reply:
Users don't see `tau_var`, though, unless they're really digging in to the implementation. I agree with Dan that just using `tau_constraint` is better.

Reply:
Users will see the name `tau_var` if they are using TensorBoard to inspect the weights, or doing something like `[weight.name for weight in layer]`. In this case they will see `[initial_level, tau_var]` on one hand but then `[initial_level_constraint, tau_constraint]` on the other hand. So I'm not sure I see one as better than the other necessarily. But if there's no ideal solution then yeah I'm fine with either. Just explaining why I went this way rather than the other.

Reply:
We could rename `tau` to `tau_initializer` and then `tau_var` to `tau`. That'd be more consistent with Keras conventions (the current scheme was chosen to be more consistent with Nengo conventions, but we're trying to be closer to Keras in this repo).

Reply:
Would the rename affect backwards compatibility of saved models/weights? If not this sounds like a good idea to me as it avoids this naming difference between `tau` and `tau_var` altogether. The main drawback is users would need to update how they've been calling the layer from `tau=` to `tau_intiializer=`. But if that's all ok would this be a separate PR?

Reply:
I'm fairly certain that the default format (SavedModel) will be fine as long as the order of the variables matches, which it will. I think the HDF5 format is a bit more fragile, so if someone has explicitly opted into that they might have problems, but I think that's a small enough disruption (especially considering we're in alpha, where breaking changes are expected).  

The `tau` -> `tau_initializer` change is a bit more drastic. But e could deprecate it more gradually (keeping around `tau`, which would set `tau_initializer=tf.initializers.constant(tau)`). But again given that we're in alpha I think it'd also be fine to go straight to the breaking change. I don't have a strong preference either way.

Reply:
Added a commit with this renaming. I went with kind of a middle ground. I didn't keep a deprecated ``tau`` parameter around, but I did make it so that you can pass in floats. So if people were using it as a positional argument, like `Lowpass(0.1)` , that will continue to work with no modifications needed.